docs/proposals/pvcs-smaller-than-1gi.md

```diff
@@ -0,0 +1,706 @@
+        // ...
+        vg, err := command.FindVolumeGroup(dc.VolumeGroup)
+        // ...
+        oc := s.ocmapper.LvcreateOptionClass(req.LvcreateOptionClass)
++/-     requested := req.GetSize()
++/-     free := int64(0)
+        var pool *command.ThinPool
+        switch dc.Type {
+        case TypeThick:
+            free, err = vg.Free()
+            // ...
+        case TypeThin:
+            pool, err = vg.FindPool(dc.ThinPoolConfig.Name)
+            // ...
+            tpu, err := pool.Free()
+            // ...
++/-         free = int64(math.Floor(dc.ThinPoolConfig.OverprovisionRatio*float64(tpu.SizeBytes))) - tpu.VirtualBytes
+        default:
+            // technically this block will not be hit however make sure we return error
+            // in such cases where deviceclass target is neither thick or thinpool
+            return nil, status.Error(codes.Internal, fmt.Sprintf("unsupported device class target: %s", dc.Type))
+        }
+
++/-     if free < requested {
+            log.Error("no enough space left on VG", map[string]interface{}{
+                "free":      free,
+                "requested": requested,
+            })
+            return nil, status.Errorf(codes.ResourceExhausted, "no enough space left on VG: free=%d, requested=%d", free, requested)
+        }
+        // ...
+        var lv *command.LogicalVolume
+        switch dc.Type {
+        case TypeThick:
+            lv, err = vg.CreateVolume(req.GetName(), requested, req.GetTags(), stripe, stripeSize, lvcreateOptions)
+        case TypeThin:
+            lv, err = pool.CreateVolume(req.GetName(), requested, req.GetTags(), stripe, stripeSize, lvcreateOptions)
+        default:
+            return nil, status.Error(codes.Internal, fmt.Sprintf("unsupported device class target: %s", dc.Type))
+        }
+        // ...
+        return &proto.CreateLVResponse{
+            Volume: &proto.LogicalVolume{
+                Name:     lv.Name(),
++/-             Size:     lv.Size(),
+                DevMajor: lv.MajorNumber(),
+                DevMinor: lv.MinorNumber(),
+            },
+        }, nil
+    }
+```
+
+### Dealing with less than 1Gi storage in the TopoLVM scheduler prioritization
+
+Currently, the scheduler uses the following scoring methodology for a node: `min(10, max(0, log2(capacity >> 30 / divisor)))`
+Note that the actual value of the capacity is stored in an annotation of the form `topolvm.GetCapacityKeyPrefix()+dc`,
+and _more importantly_, contains the value of the capacity in bytes.
+
+Our simplest suggestion here, in order to not break anything, is to clearly mark the existing capacity calculation as inaccurate and introduce
+a new lower bound for less than 1Gi storage in the method `capacityToScore`:
+
+```go
+    func capacityToScore(capacity uint64, divisor float64) int {
+        // avoid logarithm of zero, which diverges to negative infinity.
+        if capacity == 0 {
+            return 0
+        }
+    
+    return score
+}
```

Comment:
Did you forget to change this?

Reply:
Yup youre right, I adjusted this to reflect the comment above